File: app/src/main/java/me/tatarka/betterfragment/sample/dagger/FragmentKey.java
package me.tatarka.betterfragment.sample.dagger;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import dagger.MapKey;
import me.tatarka.betterfragment.Fragment;

@MapKey
@Retention(RetentionPolicy.RUNTIME)
public @interface FragmentKey {
    Class<? extends Fragment> value();
}

File: app/src/main/java/me/tatarka/betterfragment/sample/dagger/FragmentKey.kt
package me.tatarka.betterfragment.sample.dagger

import dagger.MapKey
import me.tatarka.betterfragment.app.Fragment
import kotlin.annotation.AnnotationRetention.RUNTIME
import kotlin.reflect.KClass

@MapKey
@Retention(RUNTIME)
annotation class FragmentKey(val value: KClass<out Fragment>)